### ISSUE TITLE
V2 Bom questions

### ISSUE THREAD

**Issue by tuctboh** [AndersBNielsen/Relatively-Universal-ROM-Programmer#9]
Hi,

Trying to get this into JLCPCB. The board was fine and I could make some changes to colors/etc, but theres BOM issues -

C11 - It says the value is NC, does this mean its not actually needed?
J5/J6 - Having an issue finding "Connector_PinSocket_2.54mm:PinSocket_1x04_P2.54mm_Vertical"
J7 - Says its not in the CPL file so they don't know how to place it
U5 - http://ww1.microchip.com/downloads/en/DeviceDoc/25022B.pdf isn't a valid url. Is the 232-1285-00-0602J the only one that can be used, they're saying its $65 each and not in stock.

Thanks, Tuc

**Comment by tuctboh** [comment]
Would it also be possible to get a BOM/CPL with the bottom items populated too?  Tnx

**Comment by AndersBNielsen** [comment]
There are no bottom items to be populated except for the pinheaders(which I suggest doing by hand because it's easy and it makes the board so much cheaper), so you can simply skip those. U5 is the main socket, so also a "do not place". 
Be sure to deselect the headers for assembly. 

Let me know if that's not enough info to get you going :)

**Comment by tuctboh** [comment]
> There are no bottom items to be populated except for the pinheaders(which I suggest doing by hand because it's easy and it makes the board so much cheaper), so you can simply skip those. U5 is the main socket, so also a "do not place". Be sure to deselect the headers for assembly.
> 
> Let me know if that's not enough info to get you going :)

Thanks. But I'm still looking to know -

C11 - The value says NC, does this mean its not added? It's showing up as not able to source.
J5/J6 - Do you have a better part number or description for this?
J7 - Not in the CPL file, so they don't know how to place it.
U5 - They're saying its out of stock, and its 63$ a piece. Is there a better replacement for this?

 Bottom pins+U5 - I'm not that great a solderer, so was hoping to get them to build a complete unit with no soldering needed. Is it possible to create a BOM for that.

Thanks, Tuc

**Comment by AndersBNielsen** [comment]
NC means “No connect” aka don’t place.

The other pin headers you mention are debugging and not strictly needed.

It’s not that easy to have JLC solder the pins for “A1” which is the shield
pins going into the Arduino.
You might want to contact them about it - it might be expensive.

U5 is the main socket and it’s up to you if you want a generic 15.24mm wide
32 pin socket or a ZIF.
JLC don’t have ZIFs so you’ll have to send it to them.

**Comment by tuctboh** [comment]
NC - Ok, thanks.
J5/J6 - Gotcha, but still, would rather have a COMPLETE unit incase there is a time I want to debug something stupid I did. ;)
A1 - Thanks for the heads up, I'll talk to them about it. I still would need the CPL and BOM for it.
U5 - SIGH. Ok, thanks. $63 a pop, that's unreal. (Though at 200 it goes down to 28. ;) ). 

I appreciate you putting up with this. I'm trying to come up with the order as "Ready to slap onto an Arduino Uno". (Except for the INO).

Thanks, Tuc

**Comment by AndersBNielsen** [comment]
I don’t know what you’re talking about with U5 :) It’s just a socket that
has a “U” designator because of the schematic - costing cents.

I’ll give it an update with LCSC part numbers for the headers.

**Comment by tuctboh** [comment]
So this is where things stand -

A) Uploading BOM/CPL says there is no J7 in the CPL. I realize this is a test header, but maybe in the "EVERYTHING" BOM/CPL would be nice to have it added.
B) R1,R2,R3,R4,R6,R7,R8,R9,R12,R14,R15,R19,R20,R21,R23,R24,R25,R29,R30,R31 - They're saying that the description doesn't match the part (0603WAF2202T5E) but it seems fine
C) R5,R16 - Same (0603WAF2703T5E) but also seems fine
D) R10,R11 - Same (0603WAF1001T5E) but seems fine
E) J4 - Seems an issue now, says its 1) Out of stock 2) Doesn't think its coming up with the right piece (1125-1103S0S113CR01)
F) C11 - Doesn't like the NC, is there some other way to identify it as not connected or just removed?
G) J5/J6 - I realize this is a test header, but again for the EVERYTHING BOM/CPL would be nice to have it added.
H) U5 - It didn't find the part. It looks like its looking for the 3M 232-1285-00-0602J. If I hit "Search" and put that in, it tells me that "*For qty≤5,charges a fixed fee $315.1905 " which means $63.0380 a piece! I can't seem to find anything in stock thats a 32P ZIF 2.54" . I clicked "Pre-Order" and now realize I can get them to order them, which is a bit odd but ok.  I went back to J4 to see what I can do, and because its an Inventory Shortage I can only substitute something from them, they don't give me the option to pre-order.
I) For the underside, it would be really nice if those could also be added to the "EVERYTHING" BOM/CPL. (A1 you called it I think)

Sorry I'm being a total n00b, but really interested in having this come pre-assembled and ready to go!

Thanks so much for your patience and help!

**Comment by AndersBNielsen** [comment]
You've got it wrong about U5.  It's a socket but I've used the SST part as an example of what could go into the socket and template for the pinout. I've updated the BOM accordingly in the rev2 folder to a 32 pin DIP socket - you can optionally put a ZIF socket on top of that with no soldering if you get the right kind. If you want JLC to solder a ZIF for you, you will have to source it yourself, and ship it to them. 

The resistors are the basic ones and they change from time to time. Usually JLC has no trouble picking out the correct ones. Make sure you compare everything matches the schematic and correct footprint either way. 

None of the headers you mention are strictly required. If the one it defaults two is stuck as out of stock you can select a random part in stock and then deselect it. 

I will not be adding extra headers to the BOM or CPL - it's messy and this project is supposed to come without those presoldered anyway. 
If you want to make the change yourself you will have to add the 2 x 8 pin, 1 x 10 pin, and 1 x 6 pin header - either "on top" of A1 and connected to the correct nets in the schematic, or by removing A1 and redoing all the connections and the board outline. 
Maybe you can get away with adding them unconnected and exporting the CPL but DRC will probably complain.

**Comment by tuctboh** [comment]
Hi,

Thanks for the reply. I reached out to someone on Fiverr and got them to do what I needed, thanks. I put the order into PCBWay since they were a lot easier to deal with about this than JLC. I should have 5 of these coming soon and they'll be available for sale either as just a hat or complete with an Arduino. Look forward to it!

**Comment by tuctboh** [comment]
Hi,

After taking a while to source the parts, the first attempt has been made. Its 3am here and I can't sleep so I'm posting but will take a closer look and compare to yours when I'm fully awake.

What do you think? Anything look obvious? They're always very concerned about the LEDs, is there a "standard" you followed (Anode always on top or left, etc) to verify? Otherwise, the jumpers on the bottom that don't interface to the Arduino concern me, I'm not sure there will be clearance. Any comments are appreciated.

Of course, I'll also need to test this. Do you have a well known chip and code that could be used as an example to see that it programs and reads? 

THANKS!

![Pls confirm the orientation for the LED (3)](https://github.com/user-attachments/assets/f08b115e-5c70-478e-abec-0bc1edcf4376)
![Pls confirm the orientation for the LED (2)](https://github.com/user-attachments/assets/46c8d70b-4ac7-4445-950f-bc16806f57e8)
![Pls confirm the orientation for the LED (1)](https://github.com/user-attachments/assets/1a81c0a3-18e1-457a-9a4b-abb50fa31fac)
![Pls confirm the orientation for U5](https://github.com/user-attachments/assets/ee099f5f-5fba-4e55-9477-812c7b9f9703)
![1 (5)](https://github.com/user-attachments/assets/06794e00-ec92-4d6e-83d6-91bf77bb1000)
![1 (4)](https://github.com/user-attachments/assets/42331708-a6e9-4a47-a164-aead06be08de)
![1 (3)](https://github.com/user-attachments/assets/028cc45e-0f42-4aa1-a89e-94dc5249d88d)
![1 (2)](https://github.com/user-attachments/assets/9200a5cd-14f4-4fa2-92b7-fdeb2c4e7a30)
![1 (1)](https://github.com/user-attachments/assets/c00dac6b-817b-4574-8fc7-2bce5c03dfeb)

**Comment by tuctboh** [comment]
Forgot to @AndersBNielsen you.

**Comment by AndersBNielsen** [comment]
@tuctboh  Looks reasonable. It looks like the LED's are in the correct position to me. You will have to check the PCB in kiCAD, netlist or position file if in doubt about component orientation.  

Like I mentioned before J1, J2 and J3 should not be mounted unless you specifically need them. 
If you are using a controller with a USB-B socket, you will need to isolate J1 to avoid shorting it to the B socket. A piece of kapton or electrical tape will usually do. 

If you're using an Arduino Uno there's also no point in mounting J4. 
J6 is usually only used as test points and shouldn't have be mounted either. 

I also see you've started mounting the SAO header on the wrong side of the board. I have no idea if that will be in the way of anything underneath. Probably - it certainly won't be usable on the bottom. 

Looks like you have a lot of headers soldered you don't need, but haven't finished soldering the Arduino pinout headers. 

Unless the SAO header is in the way of something, it'll probably work just fine. 

For a working unit, all you actually need are the Arduino pinout headers and the main socket.

**Comment by tuctboh** [comment]
@AndersBNielsen  - Thanks so much for looking it over and all the mentions -

Additional un-necessary headers - They're there, might as well fill the holes. ;)
J4 - Might use the 65duino, so wanna keep options open.
J1 - THANKS! Very good information, will handle it with tape
SAO - Yea, I thought that looked wrong. It wasn't in the CPL and when we added it it was mistakenly marked bottom instead of top. Will get this resolved, THANKS
Finished soldering - Yea, this isn't finished, things were just in place, especially U5 since they weren't aware which way the ZIF needed, and if things were right otherwise. 

I've given them feedback and they'll tweak it and then finish all the soldering. THANKS AGAIN!

**Comment by AndersBNielsen** [comment]
You bet! Happy you’re getting somewhere :)
The SAO should also be female.

**Comment by tuctboh** [comment]
@AndersBNielsen - FEMALE?! UGH. :-/ Thanks.

Also found out why it ended up on the bottom. Apparently my tech re-generated the CPL from your source files and it's apparently marked bottom there. Might want to correct it there. I saw you had a "Bottom" file you deleted back in October, maybe the changes to the source didn't get made at the same time.

**Comment by AndersBNielsen** [comment]
The footprint comes from the SAOs that plugin to it - hence the footprint
marked as bottom.
But again - it’s only supposed to be assembled manually if you have a SAO
to plugin or want to use the pins other ways. And some people do.

Did you have it made in time for rev 2.1 or is it a 2.0 build?

**Comment by tuctboh** [comment]
@AndersBNielsen - I didn't realize it had a keyed socket. (I didn't know SAO was a thing) I already approved just a 2x3 pin header socket. Again, want to keep options open and do it now so it doesn't need to be touched otherwise for any reason.

Rev 2.1? I don't see a rev 2.1 in github. Where is that?

**Comment by tuctboh** [comment]
@AndersBNielsen - ?? ^^ ?? - Also, is there a discord to support this project?

**Comment by AndersBNielsen** [comment]
The SAO socket is meant *either* as a SAO socket or for individual use cases - like if you wanted to use it as a secondary UART. If you don't have a SAO or a use case, you might shooting yourself in the foot by mounting something. Same for the debugging headers, since they're generally meant as probe points that *can* be used to mount headers for very special cases. 

There's an invite to the Hackerspace Clubhouse clubhouse Discord server in any somewhat recent video of mine from the last year (if I post here I won't update if it expires) - here's one: https://www.youtube.com/watch?v=LWYeSAVNEWM

2.1 was in the "Fused" branch - I've finished testing it, so I just merged to main. 

This the original BOM questions have been covered, files have been updated, so I'm closing this issue. 

You're welcome to join the Discord and ask tons of questions like this :)
